LegendaryExplorer/LegendaryExplorer/Tools/PackageEditor/Experiments/PackageEditorExperimentsO.cs

```diff
@@ -490,5 +551,216 @@ class PackageEditorExperimentsO {
+                {
+                    PropertyCollection props = new PropertyCollection();
+                    if (isBMO) { props.Add(GenerateExpressionGUID()); }
+                    props.Add(new NameProperty(parameter.GetProp<NameProperty>($"{(isBMO ? "nName" : "ParameterName")}").Value,
+                        $"{(isBMO ? "ParameterName" : "nName")}"));
+                    props.Add(new ObjectProperty(parameter.GetProp<ObjectProperty>($"{(isBMO ? "m_pTexture" : "ParameterValue")}").Value,
+                        $"{(isBMO ? "ParameterValue" : "m_pTexture")}"));
+                    TextureValues.Add(new StructProperty($"{(isBMO ? "TextureParameterValue" : "TextureParameter")}", props));
+                });
+            }
+
+            if (vectorProperty != null)
+            {
+                vectorProperty.ToList().ForEach(parameter =>
+                {
+                    PropertyCollection props = new();
+                    PropertyCollection color = new();
+                    color.Add(parameter.GetProp<StructProperty>($"{(isBMO ? "cValue" : "ParameterValue")}").GetProp<FloatProperty>("R"));
+                    color.Add(parameter.GetProp<StructProperty>($"{(isBMO ? "cValue" : "ParameterValue")}").GetProp<FloatProperty>("G"));
+                    color.Add(parameter.GetProp<StructProperty>($"{(isBMO ? "cValue" : "ParameterValue")}").GetProp<FloatProperty>("B"));
+                    color.Add(parameter.GetProp<StructProperty>($"{(isBMO ? "cValue" : "ParameterValue")}").GetProp<FloatProperty>("A"));
+                    StructProperty ParameterValue = new("LinearColor", color, $"{(isBMO ? "ParameterValue" : "cValue")}", true);
+                    if (isBMO) { props.Add(GenerateExpressionGUID()); }
+                    props.Add(ParameterValue);
+                    props.Add(new NameProperty(parameter.GetProp<NameProperty>($"{(isBMO ? "nName" : "ParameterName")}").Value,
+                        $"{(isBMO ? "ParameterName" : "nName")}"));
+                    VectorValues.Add(new StructProperty($"{(isBMO ? "VectorParameterValue" : "ColorParameter")}", props));
+                });
+            }
+
+            if (scalarProperty != null)
+            {
+                scalarProperty.ToList().ForEach(parameter =>
+                {
+                    PropertyCollection props = new();
+                    if (isBMO) { props.Add(GenerateExpressionGUID()); }
+                    props.Add(new NameProperty(parameter.GetProp<NameProperty>($"{(isBMO ? "nName" : "ParameterName")}").Value,
+                        $"{(isBMO ? "ParameterName" : "nName")}"));
+                    props.Add(new FloatProperty(parameter.GetProp<FloatProperty>($"{(isBMO ? "sValue" : "ParameterValue")}").Value,
+                        $"{(isBMO ? "ParameterValue" : "sValue")}"));
+                    ScalarValues.Add(new StructProperty($"{(isBMO ? "ScalarParameterValue" : "ScalarParameter")}", props));
+                });
```

Comment:
This should be a regular foreach loop. There's no point in converting it into a list and then calling a ForEach function on it.

Reply:
Got it